### ISSUE TITLE
rocket_get_home_path(): use of rtrim() breaks file generation (minify, combine, remove query strings)

### ISSUE THREAD

**Issue by glueckpress** [wp-media/wp-rocket#771]
## Issue and solution
In what probably is an edge case scenario, the use of [`rtrim()` inside of `rocket_get_home_path()`](https://github.com/wp-media/wp-rocket/blob/v2.11.6/inc/functions/formatting.php#L320) can result in broken paths.

Replacing:
```php
$home_path = rtrim( $home_path, $wp_path_rel_to_home );
```
with: 
```php
$home_path = preg_replace( '/' . preg_quote( $wp_path_rel_to_home, '/' ) . '$/', '', $home_path );
```
did the trick.

---

## Support case history
In a recent support case, **features like _Minify_ and _Remove Query Strings_ that required file generation didn’t produce any files**, while cache files were generated correctly.

The website in question was a **subdirectory install** where `www.example.com` would be the home page, but `www.example.com/wordpress` would be the WordPress dashboard.

There seemed to be some **unexpected handling of paths going on on the server**. A URL like:
```
https://www.example.com/wordpress/wp-content/index.php
```
could be opened just fine in the browser, but when I dumped `__FILE__` from inside of the index file, it would produce this path pattern:
```
/web/htdocs/www.example.com/home/wordpress/wp-content/index.php
```
Curiously, the extra `home/` part **didn’t physically exist on the server**; in my FTP client, I would visit:
```
/web/htdocs/www.example.com/wordpress/wp-content/index.php
```

After quite an amount of debugging, I found that from inside `rocket_get_home_path()` this line:

https://github.com/wp-media/wp-rocket/blob/2bc483d9a088adddcf5b9a2983bf2c476a8283a4/inc/functions/formatting.php#L316

would log as: 
```
'/web/htdocs/www.example.com/home/wordpress/'
```

This:

https://github.com/wp-media/wp-rocket/blob/2bc483d9a088adddcf5b9a2983bf2c476a8283a4/inc/functions/formatting.php#L319

would log as: 
```
'wordpress/'
```

And this:

https://github.com/wp-media/wp-rocket/blob/2bc483d9a088adddcf5b9a2983bf2c476a8283a4/inc/functions/formatting.php#L320

would log as:
```
'/web/htdocs/www.example.com/hom'
```
while it obviously should have been:
```
'/web/htdocs/www.example.com/home'
```

Replacing `rtrim()` with `preg_replace()` as advised by @Tabrisrp and @arunbasillal fixed the issue.

**Comment by arunbasillal** [comment]
Note to self: Exact code for `preg_replace()`

`preg_replace( '/' . preg_quote( $wp_path_rel_to_home, '/' ) . '$/', '', $home_path );`

Todo: test this on all OS.

**Comment by glueckpress** [comment]
@Tabrisrp / @arunbasillal Could this pose a similar issue?

https://github.com/wp-media/wp-rocket/blob/2bc483d9a088adddcf5b9a2983bf2c476a8283a4/inc/front/enqueue.php#L63

**Comment by Tabrisrp** [comment]
@glueckpress it should not, in this case the characters mask to be deleted is clearly defined

**Comment by xbuzz** [comment]
This change, shipped in `2.11.7` has reintroduce [this bug](https://github.com/wp-media/wp-rocket/issues/728).
